config-processor.js

```diff
@@ -0,0 +1,156 @@
+ * @returns {PluginList} The filtered list of plugins
+ */
+// eslint-disable-next-line max-statements, complexity
+function applyIncludeList(pluginManager, orgPlugins, repoIncludeList, onInvalidPlugin) {
+  const plugins = deepClone(orgPlugins);
+  for (const pluginToInclude of repoIncludeList) {
+    const pluginName = typeof pluginToInclude === 'string' ? pluginToInclude : pluginToInclude.plugin;
+    const pluginInstance = pluginManager[pluginName];
+    if (!pluginInstance) {
+      onInvalidPlugin && onInvalidPlugin(pluginName);
+      // eslint-disable-next-line no-continue
+      continue;
+    }
+
+    if (typeof pluginToInclude === 'string' &&
+      !plugins.find(p => p === pluginToInclude ||
+        (/** @type {Plugin} */(p).plugin && /** @type {Plugin} */(p).plugin === pluginToInclude))) {
+      // If the entry in the include list is just a string, we can simply add it if it's not present.
+      plugins.push(pluginToInclude);
+    } else if (typeof pluginToInclude === 'object' && pluginToInclude.plugin) {
+      // If the entry in the include list is an object, we have to be more careful.
+      const existingPlugin = plugins.find(
+        p => p === pluginToInclude.plugin ||
+        (/** @type {Plugin} */(p).plugin && /** @type {Plugin} */(p).plugin === pluginToInclude.plugin));
+      if (!existingPlugin) {
+        // If it's not in the existing list, just add it directly
+        plugins.push(pluginToInclude);
+      } else if (typeof existingPlugin === 'string' || !existingPlugin.config) {
+        // If it is in the existing list but only as a string, we can just replace the string with the object.
+        // Same story if the existing plugin has no config field set.
+        const idx = plugins.indexOf(existingPlugin);
+        plugins.splice(idx, 1, pluginToInclude);
+      } else {
+        // If it is in the existing list as an object, and the plugin to include has config specified, we have to merge
+        // the config
+        existingPlugin.config = pluginInstance.mergeConfig(existingPlugin.config, pluginToInclude.config);
+      }
+    }
+  }
+
+  return plugins;
+}
+
+/**
+ * Apply the exclude list of plugin names and return the filtered plugin list
+ *
+ * @param {PluginList} orgPlugins The list of plugins configured on the org
+ * @param {string[]} repoExcludeList A list of plugin names to exclude
+ * @returns {PluginList} The filtered list of plugins
+ */
+function applyExcludeList(orgPlugins, repoExcludeList) {
+  return orgPlugins.filter(plugin => {
+    if (typeof plugin === 'string') {
+      return !~repoExcludeList.indexOf(plugin);
+    } else if (plugin.plugin) {
+      return !~repoExcludeList.indexOf(plugin.plugin);
+    }
+
+    // Default to leaving things in the plugin list
+    return true;
+  });
+}
+
+// For unit testing
+processConfig._applyIncludeList = applyIncludeList;
+processConfig._applyExcludeList = applyExcludeList;
```

Comment:
Worth considering for future non-blocking improvements: a stand-alone library to help probot apps read from a central org configuration (e.g. `probot-dot-github` or whatnot). Could definitely see this being more general once we learn best practices in consuming `.github` content with `pullie`.

Reply:
They had [an extension](https://github.com/probot/probot-config) to do this that's now built into Probot. But it requires explicitly extending a shared config, and that doesn't really solve for the problem I'm dealing with here (many existing config files).